### ISSUE TITLE
Indentation rule fails on every line after inline lambda in chain

### ISSUE THREAD

**Issue by FWDekker** [detekt/detekt#4387]
## Observed Behavior
If a lambda argument is passed to a function but is not the last argument, and that function is then chained, then Detekt will say that _every line_ after that line is incorrectly indented, thinking that those lines each miss four spaces.

## Expected Behavior
Detekt works.

## Steps to Reproduce
Consider the following code sample, wherein the function `doWithLambda` takes a lambda that is not the last argument. Then, in `theError`, an single-line inline lambda is passed to `doWithLambda` and its return value is chained with `toDouble`.

```kotlin
class DetektSample {
    fun doWithLambda(a: () -> Int, b: Int): Int {
        return 1
    }

    fun theError() {
        doWithLambda({ 4 }, 5).toDouble()
    }
}
```

Running `./gradlew detekt` gives me the following weighted issues:
```
[redacted]/DetektSample.kt:11:1: Unexpected indentation (4) (should be 8) [Indentation]
[redacted]/DetektSample.kt:12:1: Unexpected indentation (0) (should be 4) [Indentation]
[redacted]/DetektSample.kt:13:1: Unexpected indentation (0) (should be 4) [Indentation]
```

The following changes to the code sample circumvent the issue:
* Store the output of `doWithLambda` in a variable, and then call `toDouble` on that on a separate line.
* Swap the arguments of `doWithLambda` so that the lambda is the last argument. It does not matter if the lambda is in the parentheses or not.
* Store the lambda in a separate variable and then pass it to `doWithLambda`.
* Reformat the invocation to:
  ```kotlin
  doWithLambda(
      {
          4
      },
      5
  ).toDouble()
  ```

Putting a `return@doWithLambda` inside the lambda does not resolve the issue.

## Context
I'm just trying to run Detekt.

## Your Environment

* Version of detekt used: 1.19.0
* Version of Gradle used (if applicable): 7.3.2
* Operating System and version: Debian 11
* Link to your project (if it's a public repository): The error occurs on [line 114 of `RandomnessIcons.kt`](https://github.com/FWDekker/intellij-randomness/blob/5ba44c067a6e8d4d7c90c9f4265701f9d43e06bf/src/main/kotlin/com/fwdekker/randomness/RandomnessIcons.kt#L114); you have to manually change the Detekt version from 1.18.1 to 1.19.0

**Comment by BraisGabin** [comment]
This seems like an error in ktlint. We just embed ktlint in the formatting rules. Can you reproduce it using ktlint directly?

**Comment by FWDekker** [comment]
Yes, I can reproduce it using ktlint without detekt.

**Comment by BraisGabin** [comment]
Could you open an issue there? There is little we can do here.

**Comment by FWDekker** [comment]
Done! I've opened pinterest/ktlint#1320

**Comment by BraisGabin** [comment]
Thanks! closing this one :)
